src/clientSideScene/sceneEntities.ts

```diff
@@ -510,32 +435,184 @@ export class SceneEntities {
   ) => {
     await kclManager.updateAst(modifiedAst, false)
     await this.tearDownSketch({ removeAxis: false })
-    this.setupSketch({ sketchPathToNode, forward, up, position: origin })
+    await this.setupSketch({
+      sketchPathToNode,
+      forward,
+      up,
+      position: origin,
+      maybeModdedAst: kclManager.ast,
+    })
+    this.setupSketchIdleCallbacks(sketchPathToNode)
   }
-  setUpDraftArc = async (
+  setUpDraftSegment = async (
     sketchPathToNode: PathToNode,
     forward: [number, number, number],
-    up: [number, number, number]
+    up: [number, number, number],
+    origin: [number, number, number],
+    segmentName: 'line' | 'tangentialArcTo' = 'line',
+    shouldTearDown = true
   ) => {
-    await this.tearDownSketch({ removeAxis: false })
-    await new Promise((resolve) => setTimeout(resolve, 100))
-    this.setupSketch({
+    const _ast = JSON.parse(JSON.stringify(kclManager.ast))
+
+    const variableDeclarationName =
+      getNodeFromPath<VariableDeclaration>(
+        _ast,
+        sketchPathToNode || [],
+        'VariableDeclaration'
+      )?.node?.declarations?.[0]?.id?.name || ''
+    const sg = kclManager.programMemory.root[
+      variableDeclarationName
+    ] as SketchGroup
+    const lastSeg = sg.value.slice(-1)[0] || sg.start
+
+    const index = sg.value.length // because we've added a new segment that's not in the memory yet, no need for `-1`
+
+    let modifiedAst = addNewSketchLn({
+      node: kclManager.ast,
+      programMemory: kclManager.programMemory,
+      to: [lastSeg.to[0], lastSeg.to[1]],
+      from: [lastSeg.to[0], lastSeg.to[1]],
+      fnName: segmentName,
+      pathToNode: sketchPathToNode,
+    }).modifiedAst
+    modifiedAst = parse(recast(modifiedAst))
+
+    const draftExpressionsIndices = { start: index, end: index }
```

Comment:
This is a good example of how to use the setupSketch method now I think @franknoirot , 
Not 100% sure of how you plan on doing the rectangle, but assuming that the first click is add the start of the rectangle, after which we animate the 4 draft lines until the user adds their second click, here's how I'd think about using this for animating those 4 lines between clicks.

Similar to this we modify the AST to add our new segment/s, and we grab the index of the draft lines, it's dynamic for me, but I think for rectangle it's always going to be `{ start: 0, end: 3}` yeah?